src/backends/s3.rs

```diff
@@ -484,7 +486,7 @@ fn fetch_releases_from_s3(
 
     let mut current_tag = Tag::Other;
     let mut current_release: Option<Release> = None;
-    let regex = Regex::new(r"(?i)(?P<name>.+)-(?P<version>\d+\.\d+\.\d+)-.+").map_err(|err| {
+    let regex = Regex::new(r"(?i)(?P<prefix>.*/)*(?P<name>.+)-[v]{0,1}(?P<version>\d+\.\d+\.\d+)-.+").map_err(|err| {
```

Comment:
Is the prefix addition required - it looks like it's stripping off the prefix from the thing assigned to `release.name` (and this also happens for the `exe_name` but by splitting the file path instead)? I'm a little worried that this may break expected behavior, removing `/`s (but maybe it should be doing that anyway?). Mind sharing an example of what you're trying to do to help me understand? 🙂

Reply:
sure, we have paths with a prefix that includes directories, so the regex needs to strip the leading path elements so that with a prefix of `dir/abc/whatever`, the name ends up being `whatever` instead of `dir/abc/whatever`, and the other splitting needs to be there so the download name is `whatever-v0.2.0-TARGET.tgz` instead of `dir/abc/whatever-v0.2.0-TARGET.tgz` (which the file system cannot create, because it's not a real directory).

the pathing logic could probably be simplified a bunch to only do this once, but, in the current form there wasn't an obviously more efficient way.

> I'm a little worried that this may break expected behavior, removing /s (but maybe it should be doing that anyway?)

_if_ anyone had a path in their prefix on previous versions this tried to write to a file `/tmp/.../path/whatever...`which generally doesn't exist / work anyway so i _think_ this should not be a problem?

Reply:
Thanks for explaining! Yeah, I think you're right and this should be safe